contributors.py

```diff
@@ -18,76 +20,141 @@
+    ]
+
+    all_contributors = list(github_contributors.values()) + additional_contributors
+    random.shuffle(all_contributors)
+
+    outpath = export_path.format(team="all")
+    with open(outpath, "w") as outfile:
+        logger.info(f"writing about {len(all_contributors)} contributors to {outpath}")
+        json.dump({"contributors": all_contributors}, outfile, indent=2, default=dumper)
+
+
+GITHUB_REPOS = {
     "scrap": "CovidTrackerFr/vitemadose",
     "web": "CovidTrackerFr/vitemadose-front",
     "ios": "CovidTrackerFr/vitemadose-ios",
     "android": "CovidTrackerFr/vitemadose-android",
     "infra": "CovidTrackerFr/covidtracker-server",
 }
-
-
-def main(teams=SECTIONS, export_path=EXPORT_PATH):
+def get_github_contributors(teams=GITHUB_REPOS):
     contributors_by_team = {}
     for team, path in teams.items():
         logger.info(f"getting contributors for team '{team}'")
         response = client.get(f"https://api.github.com/repos/{path}/contributors")
         response.raise_for_status()
         team_contributors = response.json()
         contributors_by_team[team] = [
-            map_github_contributor(team=team, **contributor) for contributor in team_contributors
+            GithubContributor(team=team, row=contributor) for contributor in team_contributors
         ]
 
     all_contributors = {}
     for team, contributors in contributors_by_team.items():
         for contributor in contributors:
-            if contributor.id in all_contributors:
-                all_contributors[contributor.id].teams.append(team)
-                contributor.teams = all_contributors[contributor.id].teams
+            if contributor.github in all_contributors:
+                all_contributors[contributor.github].teams.add(team)
             else:
-                all_contributors[contributor.id] = contributor
-
-    for team, contributors in contributors_by_team.items():
-        outpath = export_path.format(team=team)
-        logger.info(f"writing about {len(contributors)} contributors to {outpath}")
-        with open(outpath, "w") as outfile:
-            json.dump({"contributors": contributors}, outfile, indent=2)
-
-    outpath = export_path.format(team="all")
-    with open(outpath, "w") as outfile:
-        logger.info(f"writing about {len(all_contributors)} contributors to {outpath}")
-        json.dump({"contributors": list(all_contributors.values())}, outfile, indent=2)
-
-
-def map_github_contributor(team, login, avatar_url, html_url, **kwargs):
+                all_contributors[contributor.github] = contributor
+
+    return all_contributors
+
+DEFAULT_CSV_PATH = os.path.join(os.path.dirname(__file__), './data/input/benevoles.csv')
+def get_benevoles_csv_contributors(csv_path=DEFAULT_CSV_PATH):
+  contributors = []
+  with open(csv_path, 'r') as infile:
+    csvreader = csv.DictReader(infile, delimiter=",")
+    for row in csvreader:
+      contributors.append(CsvContributor(row))
```

Comment:
Soyons _Pythonic_ :smile: 
```suggestion
    contributors = [CsvContributor(row) for row in csvreader]
```
(et tu peux même supprimer la ligne 73)

Reply:
cheers